core/src/main/java/hudson/tasks/ArtifactArchiver.java

```diff
@@ -145,6 +152,9 @@ public Object readResolve() {
         if (caseSensitive == null) {
             caseSensitive = true;
         }
+        if (flattenDirectories == null) {
+            flattenDirectories = false;
```

Comment:
Could just be a `boolean` then. This is only needed if the default isn't `false`.